### PR TITLE
Adds locale to the locale kebab-case warnings

### LINKED ISSUE #3938: Kebab-cased translation key warning when running `just p frontend i18n`
## Description

When I run `just p frontend i18n` I get the following warnings:

```
➜  openverse git:(fix/prevent-axios-error-sentry) just p frontend i18n
pnpm --filter frontend run i18n 

> @openverse/frontend@2.2.1 i18n /home/sara/projects/openverse/frontend
> pnpm i18n:create-locales-list && pnpm i18n:get-translations && pnpm i18n:update-locales


> @openverse/frontend@2.2.1 i18n:create-locales-list /home/sara/projects/openverse/frontend
> node src/locales/scripts/create-wp-locale-list

250 locales found in GP source code.
198 locales found in WP GP instance.
Successfully wrote locales list file to /home/sara/projects/openverse/frontend/src/locales/scripts/wp-locales.json

> @openverse/frontend@2.2.1 i18n:get-translations /home/sara/projects/openverse/frontend
> node src/locales/scripts/get-translations

Successfully saved English translation to en.json.
Performing bulk download.
Found kebab-cased key in translation strings: feat-name
Found kebab-cased key in translation strings: feat-state
Found kebab-cased key in translation strings: marked-licensed
Found kebab-cased key in translation strings: view-legal
Found kebab-cased key in translation strings: terms-copy
Found kebab-cased key in translation strings: creator-name
Found kebab-cased key in translation strings: read-more
Found kebab-cased key in translation strings: read-more
Found kebab-cased key in translation strings: feat-name
Found kebab-cased key in translation strings: feat-state
Found kebab-cased key in translation strings: link-api
Found kebab-cased key in translation strings: single-name
Found kebab-cased key in translation strings: creative-commons
Found kebab-cased key in translation strings: making-wordpress
Found kebab-cased key in translation strings: marked-licensed
Found kebab-cased key in translation strings: view-legal
Found kebab-cased key in translation strings: terms-copy
Found kebab-cased key in translation strings: creator-name
Found kebab-cased key in translation strings: link-api
Found kebab-cased key in translation strings: single-name
Found kebab-cased key in translation strings: making-wordpress
Found kebab-cased key in translation strings: marked-licensed
Found kebab-cased key in translation strings: view-legal
Found kebab-cased key in translation strings: terms-copy
Found kebab-cased key in translation strings: creator-name
Found kebab-cased key in translation strings: read-more
Found kebab-cased key in translation strings: read-more
Found kebab-cased key in translation strings: feat-name
Found kebab-cased key in translation strings: feat-state
Found kebab-cased key in translation strings: single-name
Found kebab-cased key in translation strings: marked-licensed
Found kebab-cased key in translation strings: view-legal
Found kebab-cased key in translation strings: single-name
Found kebab-cased key in translation strings: marked-licensed
Found kebab-cased key in translation strings: view-legal
Found kebab-cased key in translation strings: terms-copy
Found kebab-cased key in translation strings: creator-name
Found kebab-cased key in translation strings: read-more
Found kebab-cased key in translation strings: read-more
Found kebab-cased key in translation strings: feat-name
Found kebab-cased key in translation strings: feat-state
Found kebab-cased key in translation strings: marked-licensed
Found kebab-cased key in translation strings: view-legal
Found kebab-cased key in translation strings: terms-copy
Found kebab-cased key in translation strings: creator-name
Found kebab-cased key in translation strings: read-more
Found kebab-cased key in translation strings: read-more
Found kebab-cased key in translation strings: feat-name
Found kebab-cased key in translation strings: feat-state
Found kebab-cased key in translation strings: feat-name
Found kebab-cased key in translation strings: feat-state
Found kebab-cased key in translation strings: marked-licensed
Found kebab-cased key in translation strings: view-legal
Found kebab-cased key in translation strings: terms-copy
Found kebab-cased key in translation strings: creator-name
Found kebab-cased key in translation strings: link-api
Found kebab-cased key in translation strings: single-name
Found kebab-cased key in translation strings: creative-commons
Found kebab-cased key in translation strings: common-crawl
Found kebab-cased key in translation strings: making-wordpress
Found kebab-cased key in translation strings: feat-name
Found kebab-cased key in translation strings: feat-state
Found kebab-cased key in translation strings: marked-licensed
Found kebab-cased key in translation strings: view-legal
Found kebab-cased key in translation strings: terms-copy
Found kebab-cased key in translation strings: link-api
Found kebab-cased key in translation strings: single-name
Found kebab-cased key in translation strings: creative-commons
Found kebab-cased key in translation strings: common-crawl
Found kebab-cased key in translation strings: making-wordpress
Found kebab-cased key in translation strings: marked-licensed
Found kebab-cased key in translation strings: view-legal
Found kebab-cased key in translation strings: terms-copy
Found kebab-cased key in translation strings: creator-name
Found kebab-cased key in translation strings: creative-commons
Found kebab-cased key in translation strings: read-more
Found kebab-cased key in translation strings: read-more
Found kebab-cased key in translation strings: feat-name
Found kebab-cased key in translation strings: feat-state
Found kebab-cased key in translation strings: link-api
Found kebab-cased key in translation strings: single-name
Found kebab-cased key in translation strings: making-wordpress
Found kebab-cased key in translation strings: read-more
Found kebab-cased key in translation strings: creator-name
Found kebab-cased key in translation strings: link-api
Found kebab-cased key in translation strings: read-more
Found kebab-cased key in translation strings: read-more
Found kebab-cased key in translation strings: creative-commons
Found kebab-cased key in translation strings: marked-licensed
Found kebab-cased key in translation strings: view-legal
Found kebab-cased key in translation strings: terms-copy
Found kebab-cased key in translation strings: creator-name
Found kebab-cased key in translation strings: read-more
Found kebab-cased key in translation strings: read-more
Found kebab-cased key in translation strings: feat-name
Found kebab-cased key in translation strings: feat-state
Found kebab-cased key in translation strings: link-api
Found kebab-cased key in translation strings: single-name
Found kebab-cased key in translation strings: making-wordpress
Found kebab-cased key in translation strings: marked-licensed
Found kebab-cased key in translation strings: view-legal
Found kebab-cased key in translation strings: terms-copy
Found kebab-cased key in translation strings: creator-name
Found kebab-cased key in translation strings: creative-commons
Found kebab-cased key in translation strings: read-more
Found kebab-cased key in translation strings: read-more
Found kebab-cased key in translation strings: feat-name
Found kebab-cased key in translation strings: feat-state
Found kebab-cased key in translation strings: link-api
Found kebab-cased key in translation strings: single-name
Found kebab-cased key in translation strings: making-wordpress
Found kebab-cased key in translation strings: read-more
Found kebab-cased key in translation strings: read-more
Found kebab-cased key in translation strings: marked-licensed
Found kebab-cased key in translation strings: view-legal
Found kebab-cased key in translation strings: terms-copy
Found kebab-cased key in translation strings: creator-name
Found kebab-cased key in translation strings: creative-commons
Found kebab-cased key in translation strings: read-more
Found kebab-cased key in translation strings: read-more
Found kebab-cased key in translation strings: feat-name
Found kebab-cased key in translation strings: feat-state
Found kebab-cased key in translation strings: link-api
Found kebab-cased key in translation strings: single-name
Found kebab-cased key in translation strings: making-wordpress
Found kebab-cased key in translation strings: marked-licensed
Found kebab-cased key in translation strings: view-legal
Found kebab-cased key in translation strings: terms-copy
Found kebab-cased key in translation strings: creator-name
Found kebab-cased key in translation strings: feat-name
Found kebab-cased key in translation strings: feat-state
Found kebab-cased key in translation strings: link-api
Found kebab-cased key in translation strings: single-name
Found kebab-cased key in translation strings: read-more
Found kebab-cased key in translation strings: read-more
Found kebab-cased key in translation strings: creative-commons
Found kebab-cased key in translation strings: common-crawl
Found kebab-cased key in translation strings: making-wordpress
Found kebab-cased key in translation strings: marked-licensed
Found kebab-cased key in translation strings: view-legal
Found kebab-cased key in translation strings: terms-copy
Found kebab-cased key in translation strings: creator-name
Found kebab-cased key in translation strings: creative-commons
Found kebab-cased key in translation strings: link-api
Found kebab-cased key in translation strings: single-name
Found kebab-cased key in translation strings: making-wordpress
Successfully saved 199 translations.

```

@obulat is this caused by non-English translations that haven't had their keys updated? Should we reach out to the Polyglot team and see if there is a way to bulk transform existing keys to camel case so those strings get matched in our code again?

## Reproduction

1.  `just p frontend i18n`
2.  See the logged warnings

## Additional context

Marked as a high priority, interface frontend bug fix because I suppose these keys are not getting matched by our frontend, which now uses the camel case keys, meaning these translations are all getting unnecessarily ignored. @obulat if you think the prioritisation should be different, please feel free to change it.

### PR DESCRIPTION
**Pull request by obulat**
## Fixes

Fixes #3938 by @sarayourfriend

## Description

This PR adds the locale to the warnings about kebab-cased locale keys.

The locale files contain strings with keys that can be used, for instance, to insert links into them:

`license: "{readMore} about this license.` gets converted to something like `<a href="linkURL">read more</a> about this license`

The keys like `readMore` were first in `kebab-case`. Kebab-cased keys are invalid in the latest versions of Vue i18n, so we converted them to camelCase. However, converting the keys in GlotPress is not easy and takes time. To account for that and make the translations usable, #2433 added a function that converts kebab-cased keys to camelCase. It also adds a warning to keep track if we still have some kebab-cased keys. You can see them when you run `just frontend/i18n`

The warning only had the key name. This PR adds the locale to the warning to make it possible to locate the key better:

```
bg locale kebab-cased key "marked-licensed" converted to camelCase.
bg locale kebab-cased key "view-legal" converted to camelCase.
bg locale kebab-cased key "terms-copy" converted to camelCase.
bg locale kebab-cased key "creator-name" converted to camelCase.
bg locale kebab-cased key "creative-commons" converted to camelCase.
bg locale kebab-cased key "read-more" converted to camelCase.
bg locale kebab-cased key "feat-name" converted to camelCase.
bg locale kebab-cased key "feat-state" converted to camelCase.
bg locale kebab-cased key "link-api" converted to camelCase.
bg locale kebab-cased key "single-name" converted to camelCase.
bg locale kebab-cased key "making-wordpress" converted to camelCase.
ja locale kebab-cased key "marked-licensed" converted to camelCase.
ja locale kebab-cased key "view-legal" converted to camelCase.
ja locale kebab-cased key "single-name" converted to camelCase.
da locale kebab-cased key "creative-commons" converted to camelCase.
Successfully saved 199 translations.
```

## Testing Instructions

Run `just frontend/i18n` and see that the warnings have locales.

## Checklist

- [ ] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [ ] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [ ] My commit messages follow [best practices][best_practices].
- [ ] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no visible errors.
- [ ] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by AetherUnbound** [comment]
@obulat I get the following when I try to run the testing command you've supplied:

```
$ j frontend/i18n
error: Justfile does not contain recipe `i18n`.
```

`j frontend/run i18n` does appear to run the internationalization steps though

**Review by AetherUnbound — APPROVED** [review]
Changeset looks good and the command runs as expected locally 🚀

**Review by sarayourfriend — APPROVED** [review]
LGTM. I wonder if we even need a log-per-line. If we are able to automatically convert them without issue, we could have a single log at the end with a summary "Automatically fixed _x_ kebab-case keys to camel case", but just a nit-pick. Reducing the logging would make this look less like an _error_ and more like an "here's some info" log.

File: frontend/src/locales/scripts/utils.js [hunk 1]
```diff
@@ -40,36 +40,40 @@ exports.setToValue = function setValue(obj, path, value) {
...
   if (json === null) {
     return null
   }
+
+  function replacer(_, match) {
+    // Replace ###<text>### from `po` files with {<text>} in `vue`.
+    // Additionally, the old kebab-cased keys that can still be in the
+    // translations are replaced with camelCased keys the app expects.
+    // TODO: Remove `camel` and warning once all translation strings are updated.
+    // https://github.com/WordPress/openverse/issues/2438
```
Comment (dhruvkb) [inline frontend/src/locales/scripts/utils.js:60]: This can be converted into the function docstring.

**Review by dhruvkb — APPROVED** [review]
👍 The information about the locale raising the warning is good to have!

File: frontend/src/locales/scripts/utils.js (hunk 1 above)
Comment (obulat) [reply frontend/src/locales/scripts/utils.js:60]: Thank you, done

**Comment by obulat** [comment]
> LGTM. I wonder if we even need a log-per-line. If we are able to automatically convert them without issue, we could have a single log at the end with a summary "Automatically fixed _x_ kebab-case keys to camel case", but just a nit-pick. Reducing the logging would make this look less like an _error_ and more like an "here's some info" log.

I refactored this code to log a single line by default. Also, if you run `just frontend/run i18n:get-translations --verbose`, you can see all the keys grouped by locale. I used that list to go to GlotPress and suggest the updated translations  :)

**Pull request merged**